impress-org/givewp
# Refactor: update list table confirmation modals

Resolves [GIVE-2034]

## Description
This PR updates the List Table Confirmation Modals. This is the modal that is displayed when needing to confirm a default form for the campaign, any bulk actions from the bulk action selectors, or singular action from the Row Actions. These actions are for editing, deleting or setting of items from the List Tables.

The content of the modals have been updated to reflect consistencies with the new design.

A new state property to allow the Confirmation Button text to be updated by passing in a 5th property to `ShowConfirmModalContext`.

Enqueued design system for Donors table.

## Affects
- Campaign->Forms List Table
- All Forms List Table
- Donations List Table
- Donors List Table
- Subscriptions List table

## Visuals
https://www.loom.com/share/4f8512c0d0f245a38261173e90d46473?sid=473f4cde-bfc6-450f-9e4f-2342a999bd72

## Testing Instructions
- View each List Table above. 
- Check for visual discrepancies.
- Try all actions & bulk actions for each table.

## Pre-review Checklist



[GIVE-2034]: https://stellarwp.atlassian.net/browse/GIVE-2034?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
